### DISCUSSION TITLE
Upgrading gluonts to work with Python 3.12/mxnet 2.0

### DISCUSSION THREAD

**Discussion by hallka01** [awslabs/gluonts#3223] [Q&A]
I'm curious if there is any movement on getting the gluonts package to work with Python 3.12/mxnet 2.0. I am trying to upgrade my packages/docker containers to the latest version of Python (from 3.10) but have hit significant road blocks with changes to mxnet and numpy between the versions. 

Extra context, on a mac with an M1 chip.

**Comment by rickvalstar** [comment]
***** edit ***** 1.7.0.post2 has definite improvements over 1.6.0 so I upgraded using the same pip trick and util.py change.

First, I can't find a stable mxnet beyond 1.6.0.  1.7.0.post1 and 1.7.0.post2 are it.  No 2.0 as far as I can tell.

pip install mxnet==9999
ERROR: Could not find a version that satisfies the requirement mxnet==9999 (from versions: 0.11.0, 0.12.0, 0.12.1, 1.0.0, 1.0.0.post1, 1.0.0.post3, 1.0.0.post4, 1.1.0.post0, 1.2.0, 1.2.1, 1.2.1.post1, 1.3.0, 1.3.1, 1.4.0, 1.4.0.post0, 1.4.1, 1.5.0, 1.6.0, 1.7.0.post1, 1.7.0.post2)

I installed mxnet in 3,12 using this trick to get around it failing on a dependency to an ancient numpy :

pip install --upgrade --no-deps --force-reinstall mxnet==1.6.0

I'm successfully using gluonts 0.16.0 and numpy 1.26.4 (not the latest -- required by another package).  

I did have to change line 36 in .../.venv/Lib/site-packages/mxnet/numpy/util.py:

from bool = onp.bool
to bool = bool

because Numpy no longer has an np.bool. Since development on MXNet has stopped, I'll keep repairing the mxnet code until I can get away from GluonTS.
